Fix missing column separator in the file

maxpatiiuk opened the pull request.

maxpatiiuk committed:
Fix missing column separator in the file

grantfitzsimmons reviewed (changes requested):
This could have introduced an extra tab character:

![image](https://user-images.githubusercontent.com/37256050/227324232-613a3e1d-b812-4c3a-914b-bbd0512a4081.png)

maxpatiiuk committed:
Fix column's being off by one

Fixed the columns being off bug by refactoring the code (original implementation used string concatenation - bad idea - I belive I used it because I had to save on ram - needed to fit into a 2GB ram limit on that VM - funny thing is, ChatGPT says this is actually eating more ram - well, we didn't have ChatGPT back then)

> When you store the data as a nested array, each element in the outer array is a reference to a separate inner array, and each inner array contains the strings. This means that when you append a new string to an inner array, you are only modifying that specific inner array, and not creating a new string object or copying the entire array.
>
> On the other hand, if you store the data as a single string, every time you append a new string, you would need to create a new string object with the concatenated values of the existing string and the new string. This can consume a lot of memory, especially if the strings are large.

That's why programmers should avoid premature optimization, should test the impact of their changes (or stop worrying about changes if there is no noticeable impact) and read how the code is executed on the low-level to make correct tradeoffs

maxpatiiuk committed:
Improve wrong order node detection

Fixed duplicate "Skipping node with wrong rank order" messages
Made "Skipping node with wrong rank order" include more information

maxxxxxdlp commented:
Fixed duplicate "Skipping node with wrong rank order" messages
Made "Skipping node with wrong rank order" include more information
Fixed the columns being off bug by refactoring the code (original implementation used string concatenation - bad idea - I belive I used it because I had to save on ram - needed to fit into a 2GB ram limit on that VM - funny thing is, ChatGPT says this is actually eating more ram - well, we didn't have ChatGPT back then)


> When you store the data as a nested array, each element in the outer array is a reference to a separate inner array, and each inner array contains the strings. This means that when you append a new string to an inner array, you are only modifying that specific inner array, and not creating a new string object or copying the entire array.
>
> On the other hand, if you store the data as a single string, every time you append a new string, you would need to create a new string object with the concatenated values of the existing string and the new string. This can consume a lot of memory, especially if the strings are large.

That's why programmers should avoid premature optimization, should test the impact of their changes (or stop worrying about changes if there is no noticeable impact) and read how the code is executed on the low-level to make correct tradeoffs


--

To test:
Make sure this is not present in the file when selected Mollusca -> Bivalvia (notice the glued URL and `incertae sedis`):
```
https://www.marinespecies.org/aphia.php?p=taxdetails&id=105incertae sedis
```

Make sure columns are not off by one:
![image](https://user-images.githubusercontent.com/40512816/227689742-963c648c-75d0-4157-a436-5901eef4ea58.png)

maxxxxxdlp commented:
If this code works for worms, I can do the same modification for catalog of line
